src/components/SignUp/SignUp.js

```diff
@@ -0,0 +1,206 @@
+import { Component} from 'react';
+import {connect} from 'react-redux';
+/** @jsx jsx */
+import { css, jsx } from '@emotion/core';
+import MainButton from '../MainButton';
+import { signUp } from '../../redux/actions';
+
+
+class SignUp extends Component {
+  // state holding user data
+  state = {
+    user: {}
+  };
+
+  componentDidUpdate () {
+    const {user,history} = this.props;
+    if(user && user.token) {
+      history.push('/dashboard');
+    }
+  }
+
+  //  updates state upon input value change
+  handleChange = e => {
+    const { user } = this.state;
+    this.setState({
+      user: { ...user, [e.target.name]: e.target.value }
+    });
+  };
+
+  // sends http request creating the User
+  handleSubmit = () => {
+    const { user }= this.state;
+    const {signUpAct} = this.props;
+    this.setState({});
+    const api = {
+      route: 'parties',
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(user)
```

Comment:
As the `apiMiddleware` is a RESTful+JSON api client (or at least we're setting the `Content-Type` to `application/json`) the body should be stringified by the middleware itself.